clearwater-etcd/usr/bin/etcdwrapper

```diff
@@ -1,7 +1,25 @@
 #!/bin/bash
 
-# Tiny wrapper script to redirect etcd's stdout and stderr to a file and raise appropriate ENT logs
+# Wrapper script to redirect etcd's stdout and stderr to a file and raise appropriate ENT logs
+# This also uses flock(1) to ensure that only one etcd process can write the pidfile at once
+# See https://tobrunet.ch/2013/01/follow-up-bash-script-locking-with-flock/ for more on Bash file locking
+
+PIDFILE=/var/run/clearwater-etcd/clearwater-etcd.pid
+PIDFILE_LOCK=/var/run/clearwater-etcd/clearwater-etcd.pid.lock
+
+# Claim file handle 200, writing to the lock file, so we can reference it on
+# the next line
+exec 200>$PIDFILE_LOCK 
```

Comment:
That's the

```
# Allow us to write to the pidfile directory
install -m 755 -o $NAME -g root -d /var/run/$NAME && chown -R $NAME /var/run/$NAME
```

change in debian/clearwater-etcd.init.d - that;s also how we create that directory for Bono, Sprout etc.